File: frontend/src/app/models/gamedataDTOs/RuneDefinitionDTO.ts
import { StatType } from "../../enums/StatType";

export interface RuneDefinitionDTO {
    runeType: string;
    name: string;
    runeLimitGroup: string;
    defaultTier: number;
    statsPerTier: Record<string, Record<StatType, number>>;
    description: string;
}

File: backend/src/main/java/com/langleon/dsobuildsim/runes/dto/RuneDefinitionDTO.java
package com.langleon.dsobuildsim.runes.dto;

import com.langleon.dsobuildsim.common.StatType;
import com.langleon.dsobuildsim.runes.enums.RuneLimitGroup;
import com.langleon.dsobuildsim.runes.enums.RuneType;

import java.util.Map;

public record RuneDefinitionDTO(RuneType runeType, String name, RuneLimitGroup runeLimitGroup, int defaultTier, Map<Integer, Map<StatType, Double>> statsPerTier, String description) {
}
